### PR TITLE
Filter Form for numeric and boolean fields

### PR DESCRIPTION
**Pull request by ces-luannguyen**
Hi @tototrinh 
This PR is for creating a form for filtering numeric and boolean fields.
Please help me review it.

Thanks,
Luan Nguyen

### CODE REVIEW

File: client-extensions/liferay-object-collection-filter-extension/src/Components/FilterForm.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import ClayDropdown from "@clayui/drop-down";
+import ClayButton from "@clayui/button";
+import ClayForm, {ClayCheckbox} from "@clayui/form";
+import Input from "./Input";
+const FilterForm = ({fieldName, fieldType}) => {
+    return (
+        <div style={{width: '240px', margin: '10px'}}>
```
Comment (ces-huynguyen) [inline client-extensions/liferay-object-collection-filter-extension/src/Components/FilterForm.js]: Please separate style from js file.

**Review by ces-huynguyen — CHANGES_REQUESTED** [review]

File: client-extensions/liferay-object-collection-filter-extension/src/Components/FilterForm.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                    <ClayButton
+                        displayType="secondary"
+                        className="bg-light font-weight-normal form-control-select text-left w-100"
+
+                    >
+                        {fieldType}
+                    </ClayButton>
+                }
+            >
+                {fieldType == 'numeric' && (
```
Comment (kms-hoangdao) [inline client-extensions/liferay-object-collection-filter-extension/src/Components/FilterForm.js]: I think use === is better

File: client-extensions/liferay-object-collection-filter-extension/src/components/FilterForm.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import NumericField from "./NumericField";
+import BooleanField from "./BooleanField";
+import DateField from "./DateField";
+
+const FilterForm = ({fieldName, fieldType}) => {
+
+    let input = null;
+    if (fieldType == 'numeric') {
```
Comment (kms-hoangdao) [inline client-extensions/liferay-object-collection-filter-extension/src/components/FilterForm.js:9]: I think it better if we use switch case here 
`let input;

switch (fieldType) {
    case 'numeric':
        input = <NumericField fieldName={fieldName} fieldType={fieldType} />;
        break;
    case 'boolean':
        input = <BooleanField fieldName={fieldName} fieldType={fieldType} />;
        break;
    case 'date':
        input = <DateField fieldName={fieldName} />;
        break;
    default:
        // Handle the case when fieldType doesn't match any of the specified cases
        break;
}
`

File: client-extensions/liferay-object-collection-filter-extension/src/components/FilterSettings.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import FilterForm from './FilterForm'
+const FilterSettings = ({settings}) => {
+    return (
+        <div>
+            {settings && settings.fieldArray && settings.fieldArray.map((item) => (
+                <div className="d-inline-block">
```
Comment (kms-hoangdao) [inline client-extensions/liferay-object-collection-filter-extension/src/components/FilterSettings.js:7]: when you using map in React it should have key for each item

**Comment by ces-totrinh** [comment]
There were some structural changes made to the code. Could you please re-send the pull request?

**Pull request closed without merging**